WordPress/gutenberg
# [PoC] Global styles Custom CSS: use codemirror for editing

## What?
Add `codemirror` to allow autoformatting, etc. when editing custom CSS in global styles.

## Why?
Currently the plain text input with no formatting makes editing custom CSS difficult, and codemirror is already used in the core project for the CSS editor in Customizer.

Possible fix for: https://github.com/WordPress/gutenberg/issues/47945

## How?
Adds codemirror to the Gutenberg project and lazy loads it only if the custom CSS editing tab is loaded in the site editor.  

I did explore using the existing core codemirror bundle, but this would involve running script-loader.php file for the site editor which is a very outdated approach to loading js dependencies, and would also load the full codemirror dependency even if the user never accessed the custom CSS panel.

## Testing Instructions
- Go to the site editor and load the editor
- Select the global styles panel on the right
- Make sure dev tools network tab is open
- In the overflow menu top right select Additional CSS
- Make sure the module index.js file loads when you do this and that the editor loads
- Add some CSS to the editor box and make sure line numbers appear, CSS autocompletion appears N.B don't worry about text indentation, or trying to save, neither of these are working on this diff yet.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/3629020/7d447cfa-c5e7-478f-92ec-85c79833cad6
